src/clr-angular/button/button-loading/loading-button.ts

```diff
@@ -5,26 +5,84 @@
+            <span *ngSwitchCase="buttonState.LOADING" class="spinner-btn-wrapper">
+                <span @spinner class="spinner spinner-inline spinner-loading"></span>
+            </span>
+            <span *ngSwitchCase="buttonState.SUCCESS" class="spinner-btn-wrapper">
+                <span @validated class="spinner spinner-inline spinner-loading spinner-check"></span>
+            </span>
+            <span *ngSwitchCase="buttonState.DEFAULT" @defaultButton>
+                <ng-content></ng-content>
+            </span>
+        </ng-container>
     `,
-    providers: [{provide: LoadingListener, useExisting: ClrLoadingButton}]
+    providers: [{provide: LoadingListener, useExisting: ClrLoadingButton}],
+    animations: [
+        trigger("defaultButton",
+                [
+                    transition(":enter", [style({opacity: 0}), animate("200ms 100ms ease-in", style({opacity: 1}))]),
+                    transition(":leave", [style({
+                                   opacity: 0
+                               })])  // TODO: see if we can get leave animation to work before spinner's enter animation
+                ]),
+        trigger("spinner",
+                [
+                    transition(":enter", [style({opacity: 0}), animate("200ms 100ms ease-in", style({opacity: 1}))]),
+                    transition(":leave", [style({opacity: 1}), animate("100ms ease-out", style({opacity: 0}))])
+                ]),
+        trigger("validated",
+                [
+                    transition(":enter",
+                               [animate("300ms", keyframes([
+                                            style({transform: "scale(0,0)"}), style({opacity: 1}),
+                                            style({transform: "scale(1.2,1.2)"}), style({transform: "scale(.9,.9)"}),
+                                            style({transform: "scale(1,1)"})
+                                        ]))]),
+                    transition(":leave", [style({opacity: 1}), animate("100ms ease-out", style({opacity: 0}))])
+                ])
+    ]
 })
 export class ClrLoadingButton implements LoadingListener {
-    public loading: Boolean;
+    public buttonState = ClrLoadingState;
+    public state: ClrLoadingState = ClrLoadingState.DEFAULT;
 
-    startLoading(): void {
-        this.loading = true;
-    }
+    @Output("clrLoadingChange")
+    public clrLoadingChange: EventEmitter<ClrLoadingState> = new EventEmitter<ClrLoadingState>(false);
+
+    constructor(private el: ElementRef, private renderer: Renderer2) {}
+
+    updateLoadingState(state: ClrLoadingState): void {
+        this.state = state;
+
+        switch (state) {
+            case ClrLoadingState.DEFAULT:
+                this.renderer.removeAttribute(this.el.nativeElement, "disabled");
+                break;
+            case ClrLoadingState.LOADING:
+                const width = getComputedStyle(this.el.nativeElement).getPropertyValue("width");
```

Comment:
I'll let @gnomeontherun confirm, but I believe we want to wrap this in a check for server-side rendering.